### 20200123
---

Flask에서는 Sentry 적용이 간단했다.
닷넷과 마찬가지로 Init만 해주면 알아서 처리를 해주었는데 별다른 프레임워크를 사용하지 않는 Python 프로그래밍에서는 Sentry를 사용할 때, Init 이후에 오류 메시지를 보내는 처리를 또 해야한다.

```python
import sentry_sdk

# Sentry 세팅
sentry_sdk.init(
    dsn=config.SENTRY['dsn'],
    environment=config.SENTRY['environment'],
)

try:
    # ...
except Exception as ex:
    sentry_sdk.capture_message(str(ex))
```

위의 코드처럼 처리를 하면 Sentry에 정상적으로 오류 메시지를 확인하였다.
Flask에서는 아래처럼만 하면 `try - except` 를 해주지 않더라도 오류가 다 핸들링되었다.

```python
sentry_sdk.init(
    dsn=app.config["SENTRY_DSN_KEY"],
    integrations=[FlaskIntegration()],
    environment=app.config["SENTRY_DSN_ENV"],
)
```